test/cpp_extensions/csprng_extension.cu

```diff
@@ -0,0 +1,270 @@
+      aes_helper<scalar_t, UIntType<scalar_t>::type>(iter, key,
+        [from, to] __device__ (RNGValues<1>* generator) -> scalar_t {
+          uniform_real_distribution<scalar_t> uniform(from, to);
+          return uniform(generator);
+        }
+      );
+    });
+  }
+};
+
+Tensor& uniform_(Tensor& self, double from, double to, c10::optional<Generator> generator) {
+  return uniform_impl_<UniformKernel, CUDA_CSPRNG_GeneratorImpl>(self, from, to, generator);
+}
+
+// ===========================================================================================================================
+
+template<typename RNG>
+struct NormalKernel {
+  void operator()(Tensor& self, double mean, double std, c10::optional<Generator> generator) {
+    auto iter = TensorIterator::nullary_op(self);
+    const auto key_t = key_tensor(generator, aes::block_t_size, iter.device());
+    const auto key = key_t.data_ptr<uint8_t>();
+    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "normal_kernel_cuda", [&] {
+      aes_helper<scalar_t, UIntType<scalar_t>::type, 2>(iter, key,
+        [mean, std] __device__ (RNGValues<2>* generator) -> scalar_t {
+          normal_distribution<scalar_t> normal(mean, std);
+          return normal(generator);
+        }
+      );
+    });
+  }
+};
+
+Tensor& normal_(Tensor& self, double mean, double std, c10::optional<Generator> generator) {
+  return normal_impl_<NormalKernel, CUDA_CSPRNG_GeneratorImpl>(self, mean, std, generator);
+}
+
+Tensor& normal_Tensor_float_out(Tensor& output, const Tensor& mean, double std, c10::optional<Generator> gen) {
+  return normal_out_impl<NormalKernel, CUDA_CSPRNG_GeneratorImpl>(output, mean, std, gen);
+}
+
+Tensor& normal_float_Tensor_out(Tensor& output, double mean, const Tensor& std, c10::optional<Generator> gen) {
+  return normal_out_impl<NormalKernel, CUDA_CSPRNG_GeneratorImpl>(output, mean, std, gen);
+}
+
+Tensor& normal_Tensor_Tensor_out(Tensor& output, const Tensor& mean, const Tensor& std, c10::optional<Generator> gen) {
+  return normal_out_impl<NormalKernel, CUDA_CSPRNG_GeneratorImpl>(output, mean, std, gen);
+}
+
+Tensor normal_Tensor_float(const Tensor& mean, double std, c10::optional<Generator> gen) {
+  return normal_impl<NormalKernel, CUDA_CSPRNG_GeneratorImpl>(mean, std, gen);
+}
+
+Tensor normal_float_Tensor(double mean, const Tensor& std, c10::optional<Generator> gen) {
+  return normal_impl<NormalKernel, CUDA_CSPRNG_GeneratorImpl>(mean, std, gen);
+}
+
+Tensor normal_Tensor_Tensor(const Tensor& mean, const Tensor& std, c10::optional<Generator> gen) {
+  return normal_impl<NormalKernel, CUDA_CSPRNG_GeneratorImpl>(mean, std, gen);
+}
+
+// ===========================================================================================================================
+
+Generator create_CUDA_CSPRNG_Generator() {
+  return make_generator<CUDA_CSPRNG_GeneratorImpl>();
+}
+  
+void registerOps() {
+  // TODO: replace with new ops registration API
```

Comment:
Yes, we should do this! I probably won't block this PR (to pytorch/pytorch) over it, but if you end up giving this code a home somewhere else, we should switch it over, because it will be a bit harder to keep it up to date when it lives externally.